File: cgi-bin/post.py
#!/usr/bin/python3

# Import modules for CGI handling 
import cgi, cgitb 

# Create instance of FieldStorage 
form = cgi.FieldStorage()
cgitb.enable()

# Get data from fields
name = form.getvalue('name')
age  = form.getvalue('age')

print("Content-type:text/html\r\n\r\n")
print("<html>")
print("<head>")
print("<title>Hello - Second CGI Program</title>")
print("</head>")
print("<body>")
print("<h2>Hello %s %s</h2>" % (name, age))
print("</body>")
print("</html>")

File: cgi-bin/post.rb
#!/usr/bin/env ruby
require 'cgi'
cgi = CGI.new

name = cgi['name'] # => "Maria"
age = cgi['age']

body = "Hello #{name} #{age}"

len = body.size
puts "HTTP/1.1 200 OK\r\n"
puts "Content-Type: text/plain\r\n"
puts "Content-Length: #{len}\r\n\r\n"
# puts "Content-Length: 13\r\n\r\n"

puts body